compiler/plugins/target/ROCM/test/emit_debug_info.mlir

```diff
@@ -0,0 +1,52 @@
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: iree-compile --iree-hal-target-backends=rocm --iree-hip-target=gfx942 \
+// RUN:   --iree-hip-emit-debug-info \
+// RUN:   --iree-hal-dump-executable-binaries-to=%t \
+// RUN:   %s -o %t/module.vmfb
+// RUN: llvm-dwarfdump --debug-line %t/*.hsaco | FileCheck %s
```

Comment:
Would it be possible to test this still at the level of IR instead of going all the way to object files and then disassembling them?

Also, I think this test requires `shell`?